src/table.rs

```diff
@@ -565,51 +604,107 @@ impl Column {
-                let values = self
-                    .view_iter::<UInt32ArrayType, u32>(rows)
-                    .unwrap()
-                    .map(|v: &u32| IpAddr::from(Ipv4Addr::from(*v)))
-                    .collect::<Vec<_>>();
                 describe_top_n!(
-                    values.iter(),
+                    iter,
                     rows.len(),
                     desc,
-                    IpAddr,
-                    DescriptionElement::IpAddr
+                    [u8],
+                    DescriptionElement::Binary,
+                    number_of_top_n
                 );
             }
-            ColumnType::DateTime => {
+            ColumnType::IpAddr => {
                 let values = self
-                    .view_iter::<Int64ArrayType, i64>(rows)
+                    .view_iter::<UInt32ArrayType, u32>(rows)
                     .unwrap()
-                    .map(|v: &i64| {
-                        let dt = NaiveDateTime::from_timestamp(*v, 0);
-                        NaiveDateTime::new(dt.date(), NaiveTime::from_hms(dt.time().hour(), 0, 0))
-                    })
+                    .map(|v: &u32| IpAddr::from(Ipv4Addr::from(*v)))
                     .collect::<Vec<_>>();
                 describe_top_n!(
                     values.iter(),
                     rows.len(),
                     desc,
-                    NaiveDateTime,
-                    DescriptionElement::DateTime
+                    IpAddr,
+                    DescriptionElement::IpAddr,
+                    number_of_top_n
                 );
             }
+            ColumnType::DateTime => unreachable!(), // by implementation
         }
 
         desc
     }
+
+    #[must_use]
+    fn describe_datetime(
+        &self,
+        rows: &[usize],
+        time_interval: u32,
+        number_of_top_n: u32,
+    ) -> Description {
+        let mut desc = Description::default();
+
+        let time_interval = if time_interval > MAX_TIME_INTERVAL {
+            MAX_TIME_INTERVAL
+        } else {
+            time_interval
+        };
+        let time_interval = if time_interval < MIN_TIME_INTERVAL {
+            MIN_TIME_INTERVAL
+        } else {
+            time_interval
+        };
+
+        let values = self
+            .view_iter::<Int64ArrayType, i64>(rows)
+            .unwrap()
+            .map(|v: &i64| {
+                let dt = NaiveDateTime::from_timestamp(*v, 0);
+                let dt_total_second = (dt.hour() * 3600 + dt.minute() * 60 + dt.second())
+                    / time_interval
+                    * time_interval;
+                let dt_hour = dt_total_second / 3600;
+                let dt_minute = (dt_total_second - dt_hour * 3600) / 60;
+                let dt_second = dt_total_second - (dt_hour * 3600 + dt_minute * 60);
+                NaiveDateTime::new(
+                    dt.date(),
+                    NaiveTime::from_hms(dt_hour, dt_minute, dt_second),
+                )
```

Comment:
```suggestion
                let dt_total_second = *v / time_interval as i64 * time_interval as i64;
                NaiveDateTime::from_timestamp(dt_total_second);
```

Reply:
Got it. Will change soon.

Reply:
In fact, I intended that time intervals should start with 0 second of every hour. Your suggestion makes the starting time variable. What do you think?

Reply:
I’m not sure I understand it fully, could you give me an example?

Reply:
My explanation was not exactly what I wanted to told you. Sorry for confusing. (I wrote the above while I was moving.). What I intended was that the first interval of every day should start with 00:00:00. If an interval is set to 7 minutes, it will be [00:00:00, 00:06:59] ... [23:55:00, 00:01:59 of next day] ... and more from your suggestion. But, As I want the last one would be [23:55:00, 23:59:59], I separated data and time.

Reply:
And, this has something to do with that the maximum interval is one day.

Reply:
Hmm, if that's the concern, how about:
```suggestion
                let mut ts = *v / (24 * 60 * 60);
                ts += (*v - ts) / time_interval as i64 * time_interval as i64;
                NaiveDateTime::from_timestamp(ts, 0)
```
p.s. My intention was just to minimize the FnOnce, not a big deal though.
p.p.s. if we limit the time_interval within one day, would it be better to document it?

Reply:
Your calculation is clearer and better. I will update the code with yours. BTW, the reason why I separate date and time is that users want to see time series of events which belong to the same time interval of each day.

And, do mean to document it in the code?

Reply:
That makes sense. Never thought of that. I was inclined to do that at first thinking from the angle of clustering, since it might bring inequivalent weight to the last slots in the day.

As for the document, I think the user might want to know why we show it in a one day interval when he/she provided two days for example. And also we might need to document for the function `describe_datetime` since the `time_interval` it accepts is `u32`.

Reply:
To avoid weighting to the last slots, I had made enums which split a day with equal slots in the first place. Regarding in which location between frontend and structured restriction of time intervals be performed, I had thought structured would be better. But, as structured should focus on performance, it is good enough to restricted user inputs in the frontend. 

BTW, I think you mean the first line of your suggestion will be:

`let mut ts = *v / (24 * 60 * 60) * (24 * 60 * 60)`

Reply:
To be precise, `time_inteval` can be larger than a day only if it is in date units. But, the options are currently 30 seconds, 5 minutes, 10 minutes, 30 minutes, and an hour and will be verified by frontend.

Reply:
Anyway, your first suggestion can be a better option because frontend does not allow users to input improper intervals. But, I want to be prepared if users try to test some odd intervals like 7 minutes. In such a case, it's quite sure that users want to daily intervals start with 00:00:00.